src/test/java/org/opentripplanner/middleware/docs/SwaggerTest.java

```diff
@@ -0,0 +1,45 @@
+package org.opentripplanner.middleware.docs;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.middleware.OtpMiddlewareTest;
+import org.opentripplanner.middleware.utils.YamlUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.opentripplanner.middleware.docs.PublicApiDocGenerator.getSwaggerDocs;
+import static org.opentripplanner.middleware.docs.VersionControlledSwaggerUpdater.SWAGGER_CONTROLLED_FILE;
+
+/**
+ * Contains test to verify that the swagger docs are generated correctly.
+ */
+public class SwaggerTest extends OtpMiddlewareTest {
+    private static final Logger LOG = LoggerFactory.getLogger(SwaggerTest.class);
+
+    /**
+     * Verify that version-controlled doc.yaml file is up to date with the latest changes to swagger file auto-generated
+     * from the spark-swagger library (available at {@link PublicApiDocGenerator#getSwaggerDocs()}.
```

Comment:
`Verify that {@link SWAGGER_CONTROLLED_FILE} matches the latest changes to the API as generated by spark-swagger (available at {@link PublicApiDocGenerator#getSwaggerDocs()}. If there are changes to the API, run {@link VersionControlledSwaggerUpdater#main} to update the file.`

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/59/commits/b327a799b261096bd67fcaf0d680d92b1090fc2a